File: 6kyu/6kyu - Highest Scoring Word.js
// 6kyu - Highest Scoring Word

// function high(x) {
//     let alphabet = 'abcdefghijklmnopqrstuvwxyz'
//     let a = x.split(' ')
//     let b = a.map(v => v.split('').reduce((acc, curr) => acc + (alphabet.indexOf(curr) + 1), 0))
//     let c = [...b].sort((x,y) => y - x)[0]
//     let d = b.indexOf(c)
//     return a[d]
// }

// function high(x) {
//     let alphabet = 'abcdefghijklmnopqrstuvwxyz'
//     let a = x.split(' ')
//     let b = a.map(v => v.split('').reduce((t, c) => t + (alphabet.indexOf(c) + 1), 0))
//     let c = [...b].sort((x, y) => y - x)[0]
//     return a[b.indexOf(c)]
// }

function high(x) {
    let alpha = 'abcdefghijklmnopqrstuvwxyz'
    let words = x.split(' ')
    let score = words.map(v => v.split('').reduce((t, c) => t + (alpha.indexOf(c) + 1), 0))

    return words[score.indexOf(Math.max(...score))]
}

q = high('man i need a taxi up to ubud') // 'taxi'
q
q = high('what time are we climbing up the volcano') // 'volcano'
q
q = high('take me to semynak') // 'semynak'
q

File: 6kyu/6kyu - Highest Scoring Word.py
# 6kyu - Highest Scoring Word

""" Given a string of words, you need to find the highest scoring word.
Each letter of a word scores points according to its position in the alphabet: a = 1, b = 2, c = 3 etc.
You need to return the highest scoring word as a string.

If two words score the same, return the word that appears earliest in the original string.
All letters will be lowercase and all inputs will be valid. """

# def high(x):
#     words = x.split()
#     score = []
#     for w in words:
#         char_vals = []
#         for i in range(len(w)):
#             char_vals.append(ord(w[i])-96)
#         score.append(sum(char_vals))
#     return words[score.index(max(score))]


def high(x):
    words = x.split()
    score = []
    for w in words:
        score.append(sum(ord(c)-96 for c in w))
    return words[score.index(max(score))]

# def high(x):
#     return max(x.split(), key=lambda w: sum(ord(c)-96 for c in w))


q = high('man i need a taxi up to ubud'), 'taxi'
q
q = high('what time are we climbing up the volcano'), 'volcano'
q
q = high('take me to semynak'), 'semynak'
q
